src/clients/java/src/jni.zig

```diff
@@ -0,0 +1,3230 @@
+        env: *JNIEnv,
+        clazz_1: JClass,
+        clazz_2: JClass,
+    ) JBoolean {
+        return JNIEnv.interface_call(
+            env,
+            .is_assignable_from,
+            .{ clazz_1, clazz_2 },
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#toreflectedfield.
+    pub inline fn to_reflected_field(
+        env: *JNIEnv,
+        cls: JClass,
+        field_id: JFieldID,
+        is_static: JBoolean,
+    ) JObject {
+        return JNIEnv.interface_call(
+            env,
+            .to_reflected_field,
+            .{ cls, field_id, is_static },
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#throw.
+    pub inline fn throw(
+        env: *JNIEnv,
+        obj: JThrowable,
+    ) JNIResultType {
+        return JNIEnv.interface_call(
+            env,
+            .throw,
+            .{obj},
+        );
+    }
+
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#thrownew.
+    pub inline fn throw_new(
+        env: *JNIEnv,
+        clazz: JClass,
+        message: [*:0]const u8,
+    ) JNIResultType {
+        return JNIEnv.interface_call(
+            env,
+            .throw_new,
+            .{ clazz, message },
+        );
+    }
+
+    /// Returns the exception object that is currently in the process of being thrown,
+    /// or NULL if no exception is currently being thrown.
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptionoccurred.
+    pub inline fn exception_occurred(
+        env: *JNIEnv,
+    ) JThrowable {
+        return JNIEnv.interface_call(
+            env,
+            .exception_occurred,
+            .{},
+        );
+    }
+
+    /// Prints an exception and a backtrace of the stack to a system error-reporting channel,
+    /// such as stderr. This is a convenience routine provided for debugging.
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptiondescribe.
+    pub inline fn exception_describe(
+        env: *JNIEnv,
+    ) void {
+        JNIEnv.interface_call(
+            env,
+            .exception_describe,
+            .{},
+        );
+    }
+
+    /// Clears any exception that is currently being thrown.
+    /// If no exception is currently being thrown, this routine has no effect.
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#exceptionclear.
+    pub inline fn exception_clear(
+        env: *JNIEnv,
+    ) void {
+        JNIEnv.interface_call(
+            env,
+            .exception_clear,
+            .{},
+        );
+    }
+
+    /// Raises a fatal error and does not expect the VM to recover.
+    /// This function does not return.
+    /// https://docs.oracle.com/en/java/javase/17/docs/specs/jni/functions.html#fatalerror.
+    pub inline fn fatal_error(
+        env: *JNIEnv,
+        msg: [*]const u8,
+    ) void {
```

Comment:
I guess this could be 

```suggestion
    ) noreturn {
```